### PR TITLE
Add basic auth for ecs

### PR DESCRIPTION
**Pull request by mmatur**
### Description

Implements frontend basic auth for ecs provider.

This addresses the ecs part of #1465

### CODE REVIEW

File: provider/ecs/ecs_test.go [hunk 1]
```diff
@@ -348,3 +348,36 @@ func TestTaskChunking(t *testing.T) {
...
+			desc: "label existing",
+			instance: simpleEcsInstance(map[string]*string{
+				types.LabelFrontendAuthBasic: aws.String("user:password"),
+			}),
+			expected: []string{"user:password"},
+		},
+	}
+
+	for _, c := range cases {
+		c := c
```
Comment (ldez) [inline provider/ecs/ecs_test.go]: could rename `c` to `test`?

Comment (mmatur) [reply provider/ecs/ecs_test.go]: If we rename `c` to `test` the complete file will be not consistent.

https://github.com/mmatur/traefik/blob/ce55834602fc20aba66a0ecf16302729c7d7da43/provider/ecs/ecs_test.go#L333

Comment (ldez) [reply provider/ecs/ecs_test.go]: you can rename all `c` 😃

File: docs/configuration/backends/ecs.md [hunk 2]
```diff
@@ -78,17 +78,18 @@ SecretAccessKey = "123"
...
+| `traefik.frontend.passHostHeader=true`                                                                               | forward client `Host` header to the backend.                                                                         |
+| `traefik.frontend.priority=10`                                                                                       | override default frontend priority                                                                                   |
+| `traefik.frontend.entryPoints=http,https`                                                                            | assign this frontend to entry points `http` and `https`. Overrides `defaultEntryPoints`.                             |
+| `traefik.frontend.auth.basic=test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/,test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0` | Sets basic authentication for that frontend with the usernames and passwords test:test and test2:test2, respectively |
```
Comment (ldez) [inline docs/configuration/backends/ecs.md]: could you replace by
```
| `traefik.frontend.auth.basic=EXPR`                | Sets basic authentication for that frontend in CSV format: `User:Hash,User:Hash` |
```

File: provider/ecs/ecs_test.go [hunk 3]
```diff
@@ -343,8 +343,41 @@ func TestTaskChunking(t *testing.T) {
...
+			desc: "label existing",
+			instance: simpleEcsInstance(map[string]*string{
+				types.LabelFrontendAuthBasic: aws.String("user:password"),
+			}),
+			expected: []string{"user:password"},
+		},
+	}
+
+	for _, test := range cases {
+		c := test
```
Comment (timoreimann) [inline provider/ecs/ecs_test.go]: `test := test` would work as well. No need to change the variable name.

File: provider/ecs/ecs.go [hunk 4]
```diff
@@ -469,6 +470,14 @@ func (p *Provider) getFrontendRule(i ecsInstance) string {
 	return "Host:" + strings.ToLower(strings.Replace(i.Name, "_", "-", -1)) + "." + p.Domain
 }
 
+func (p *Provider) getBasicAuth(i ecsInstance) []string {
+	label := i.label(types.LabelFrontendAuthBasic)
+	if label != "" {
+		return strings.Split(label, ",")
```
Comment (timoreimann) [inline provider/ecs/ecs.go]: Do we need/also want to check that each basic auth value adheres to the `user:hash`?

File: provider/ecs/ecs_test.go (hunk 3 above)
Comment (mmatur) [reply provider/ecs/ecs_test.go]: You right. A refactoring of ECS provider has been done in #2050. I will do the change to be consistent with this refactoring

Comment (timoreimann) [reply provider/ecs/ecs_test.go]: 👍

File: provider/ecs/ecs.go (hunk 4 above)
Comment (timoreimann) [reply provider/ecs/ecs.go]: @ldez This should have been answered prior to merging.

Comment (ldez) [reply provider/ecs/ecs.go]: https://github.com/containous/traefik/blob/master/provider/docker/docker.go#L706

**Pull request merged**